src/main/java/com/joss/conductor/mobile/ConductorConfig.java

```diff
@@ -0,0 +1,379 @@
+package com.joss.conductor.mobile;
+
+import com.joss.conductor.mobile.util.Log;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class ConductorConfig {
+    private static final String DefaultConfigFile = "/config.yaml";
+
+    // Conductor properties
+    private String[] currentSchemes;
+    private int timeout = 5;
+    private int retries = 5;
+    private boolean screenshotOnFail = true;
+
+    // Appium Properties
+    private Platform platformName = Platform.NONE;
+    private String deviceName;
+    private boolean autoGrantPermissions = true;
+    private boolean noReset = true;
+    private boolean fullReset = false;
+    private String appiumVersion;
+    private String platformVersion;
+    private String appFile;
+    private String language;
+    private String locale;
+    private String orientation;
+    private String hub;
+    private String udid;
+    private String automationName;
+    private String appPackageName;
+
+    // iOS specific
+    private String xcodeSigningId;
+    private String xcodeOrgId;
+
+    // Android specific
+    private String avd;
+    private String appActivity;
+    private String appWaitActivity;
+    private String intentCategory;
+
+    public ConductorConfig() {
+        try {
+            InputStream is = this.getClass().getResourceAsStream(DefaultConfigFile);
+            if(is != null) {
```

Comment:
Can this ever be null? If so, there should probably be an `else`.

Reply:
If it is, it would should just use the "defaults" which is mostly nulls so an else shouldn't be necessary.

That said, I should check for and throw an exception in readConfig if an InputStream is null.